src/node_jsvmapi_types.h

```diff
@@ -34,4 +36,48 @@ struct napi_property_descriptor {
   void* data;
 };
 
+enum napi_valuetype {
+  // ES6 types (corresponds to typeof)
+  napi_undefined,
+  napi_null,
+  napi_boolean,
+  napi_number,
+  napi_string,
+  napi_symbol,
+  napi_object,
+  napi_function,
+};
+
+enum napi_typedarray_type {
```

Comment:
Think it would be valuable to put these into the napi_valuetype enum instead? Then napi_get_type_of_value could return the TypedArray type, as an example of use case. Not sure if useful, though. Module code likely knows when it's expecting TypedArray objects. Hmm.